### PR TITLE
[ENHANCEMENT] Add unhandled promise errors

### PR DESCRIPTION
**Pull request by seawatts**
Track.js blog just published a post about handling unhandled promises. https://trackjs.com/blog/unhandled-promises/
I thought this would be a good place to add this to the project. Thoughts?

### CODE REVIEW

File: app/instance-initializers/configure-trackjs.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ export function initialize(app) {
   const handler = new ErrorHandler(trackJs);
 
   Ember.onerror = handler.report.bind(handler);
+  window.addEventListener('unhandledrejection', (event) => {
```
Comment (jherdman) [inline app/instance-initializers/configure-trackjs.js]: Would you mind switching the function declaration style to `function(event) {`? This keeps this change in line with the rest of the code's style.

Is it possible to mock or trigger this event in a test to ensure this hook works?

Many thanks for the pull request!

Comment (seawatts) [inline app/instance-initializers/configure-trackjs.js]: Sure, might not be able to get to it until the end of the day.

**Comment by seawatts** [comment]
@jherdman do you actually know if this can be solved by using `Ember.RSVP.on('error', ...)`?

**Comment by jherdman** [comment]
@seawatts yeah, 100% and it's definitely wired up already. `Ember.onerror` is the single conduit for all of that rubbish. It doesn't help you with native promises though.

**Pull request closed without merging**